DedicatedServer/Scripts/Libs/Zrx/ModeLibs/FlagRush/UI/Modules/Radar.Script.txt

```diff
@@ -85,159 +50,471 @@ Text GetMarkersManialink() {
-				(MarkerFrame.GetFirstChild("icon") as CMlQuad).Opacity = {{{ C_PlayerMarkerIconOpacity }}};
-				(MarkerFrame.GetFirstChild("glow") as CMlQuad).Opacity = {{{ C_PlayerMarkerGlowOpacity }}};
-			} else {
-				// Determine relative position on radar
-				declare Vec3 RelativePosition = GetPlayerPosition(Player) - GetPlayerPosition(GUIPlayer);
-				declare Vec2 RelativeRadarPosition = Rotate(-GetPlayerYaw(GUIPlayer), <-RelativePosition.X, -RelativePosition.Z>) * {{{ G_Config.Zoom }}};
-
-				// Hide when outside
-				declare OffsetPercentage = ML::Distance(<0., 0.> , RelativeRadarPosition) / {{{ C_RadarBaseSize / 2 }}};
-				if (OffsetPercentage > {{{ C_MarkerFadeEnd }}}) {
-					MarkerFrame.Hide();
-					return;
+			/**
+			 * Inits all MlControls. This includes setting slider ranges and default values.
+			 */
+			Void InitMlControls() {
+				Customization_PositionXEntry.SetValueType(CMlEntry::ValueType::Ml_Real3Decimals);
+				Customization_PositionYEntry.SetValueType(CMlEntry::ValueType::Ml_Real3Decimals);
+				Customization_ScaleSlider.MinRange = 0.1;
+				Customization_ScaleSlider.MaxRange = 1.;
+				Customization_BackgroundOpacitySlider.MinRange = 0.;
+				Customization_BackgroundOpacitySlider.MaxRange = 1.;
+				Customization_DetailOpacitySlider.MinRange = 0.;
+				Customization_DetailOpacitySlider.MaxRange = 1.;
+
+				declare persistent K_RadarUserConfig RadarConfig for LocalUser;
+				Customization_ScaleSlider.Value = RadarConfig.Scale;
+				Customization_BackgroundOpacitySlider.Value = RadarConfig.BackgroundOpacity;
+				Customization_DetailOpacitySlider.Value = RadarConfig.DetailOpacity;
+			}
+
+			/**
+			 * Loads the radar config for the player.
+			 * If player doesn't have a config, then the default server config is used.
+			 */
+			 Void LoadRadarUserConfig() {
+				 declare persistent Boolean RadarConfigLoaded for LocalUser;
+				if(!RadarConfigLoaded) {
+					declare persistent K_RadarUserConfig RadarConfig for LocalUser;
+					declare persistent Text Persistent_FlagRush_RadarUserConfigJson for UserMgr.MainUser = "{{{ TL::Replace(G_Config.tojson(), "\"", "\\\"") }}}";
+					RadarConfig.fromjson(Persistent_FlagRush_RadarUserConfigJson);
+					RadarConfigLoaded = True;
 				}
-				MarkerFrame.Show();
-
-				// Determine relative rotation to the player
-				declare Real RelativeYaw;
-				if (Net_FlagRush_FlagCarrierLogin == Player.User.Login) RelativeYaw = 0.;
-				else RelativeYaw = GetPlayerYaw(Player) - GetPlayerYaw(GUIPlayer);
-
-				// Fade when close too the edge
-				declare Real Opacity = 1.0;
-				if (OffsetPercentage > {{{ C_MarkerFadeStart }}}) {
-					declare Real FadeProgress = ({{{ C_MarkerFadeEnd }}} - OffsetPercentage) / ({{{ C_MarkerFadeEnd - C_MarkerFadeStart }}});
-					Opacity = ML::Clamp(FadeProgress, 0., 1.);
+			}
+
+			/**
+			 * Persistently saves the radar config the player.
+			 */
+			Void SaveRadarUserConfig() {
+				declare persistent K_RadarUserConfig RadarConfig for LocalUser;
```

Comment:
i think this shoudln't be persistent

Reply:
> i think this shoudln't be persistent

This is not saved. Only variables for MainUser are saved.

Reply:
But ye, should not be marked as persistent here. Apparently script cache isn't a thing anymore. I gad an issue in my thinking. I Just wanted it to be preserved over reinits, so it's not reset if reinit happens during editing dialog. But a normal declare for should do the trick